### ISSUE TITLE
Feature Request - of type

### ISSUE THREAD

**Issue by mabar** [pascalduez/postcss-quantity-queries#12]
Can you add selectors for *-of-type? They are usually working more as expected than child selectors. Described on [css-tricks.com](https://css-tricks.com/the-difference-between-nth-child-and-nth-of-type/)
Something simple like :exactly-of-type would be good.

**Comment by mabar** [comment]
ping @pascalduez

**Comment by pascalduez** [comment]
Hi,

could you describe a bit more the use-cases and give a few examples?
I'm not having a clear picture about how this could be related to quantities.

**Comment by mabar** [comment]
```
<p>Foo</p>
<p>Bar</p>
```
For this code work both
 - p:nth-child(2) - select paragraph if it's second child
- p:nth-of-type(2) - select second paragraph

But when you add one element
```
<h1>Foo</h1>
<p>Bar</p>
<p>Baz</p>
```
With this code :nth-child selects first paragraph, :nth-of-type still selects second. Add one tag before paragraphs and :nth-child selects nothing.

I am using quantities in design where I sometimes need to add some spans when pseudo elements are insufficient and child selectors are breaking that.

**Comment by pascalduez** [comment]
I see what you mean now.

So out of the blue this would give something like:
```
:at-least-of-type(count) { ... }
:at-most-of-type(count) { … }
:between-of-type(start, end) { … }
:exactly-of-type(count) { … }
```

Won't be implemented before the WIP refactor for #8 is done.
